internal/runtime/client/client.go

```diff
@@ -117,6 +119,9 @@ func (c *client) IsReady() bool {
 }
 
 func (c *client) Discover(ctx context.Context, extensionConfig *runtimev1.ExtensionConfig) (*runtimev1.ExtensionConfig, error) {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info(fmt.Sprintf("Performing discovery for Extension %s/%s", extensionConfig.Namespace, extensionConfig.Name))
```

Comment:
```suggestion
	log.Info("Performing discovery for Extension")
```
In the controller the k/v pair should be already there, in warmup we should add it to the logger in the context at the beginning of the for loop (~l.104 warmup.go)